### PR TITLE
Moving platform fix

### PR DESCRIPTION
**Pull request by DelCherub**
Moving platforms have been slightly improved in regards to detecting players: it now considers the half height of the player and platform when deciding if a player is "on" the platform.

### CODE REVIEW

**Review by JunkyX1122 — APPROVED** [review]
Woomy

**Review by Joerseven — COMMENTED** [review]
I CAN"T GET ON A PLATFORM TO TEST IT BECAUSE THEY MOVE SO GODDAMN FAST WHO'S IDEA WAS IT TO MAKE THE PLATFORMS DART AROUND THE WORLD LIKE SONIC'S GRANDAD ON CRACK

File: Server/Components/ObjectOscillator.cpp [hunk 1]
```diff
@@ -15,16 +16,21 @@ ObjectOscillator::ObjectOscillator(GameObject * go, float period, float distance
 
 void ObjectOscillator::OnCollisionEnter(GameObject* other) {
     if (other->GetTag() == PLAYER) {
-        if(other->GetTransform().GetPosition().y > gameObject->GetTransform().GetPosition().y)
-        objectsOnPlatform[other] = true;
-        std::cout << "PLAYER ON PLATFORM\n";
+
+        float topOfOscillator = gameObject->GetTransform().GetPosition().y + halfHeight;
+        CapsuleVolume* capVolume = ((CapsuleVolume*)other->GetBoundingVolume());
+        float playerFeet = other->GetTransform().GetPosition().y - capVolume->GetHalfHeight();
```
Comment (Joerseven) [inline Server/Components/ObjectOscillator.cpp:22]: Shouldn't this include the radiusof the capsule? Consdering the capsule as a line segment, to my understanding the half height is half the distance from a - b that then needs to be added together with the radius. Not 100% on this so get back to me about it.

Comment (St4rman) [reply Server/Components/ObjectOscillator.cpp:22]: talked about this, getting this on main and if required will add on later. Small value anyway, shouldn't matter too much.

File: Server/States/RunningState.cpp [hunk 2]
```diff
@@ -455,7 +455,26 @@ void RunningState::BuildLevel(const std::string &levelName)
         g->GetPhysicsObject()->SetInverseMass(0.0f);
         g->GetPhysicsObject()->SetLayer(OSCILLATOR_LAYER);
 
+
+
         ObjectOscillator* oo = new ObjectOscillator(g,x->timePeriod,x->distance,x->direction,x->cooldown,x->waitDelay);
+        const CollisionVolume* vol = g->GetBoundingVolume();
```
Comment (St4rman) [inline Server/States/RunningState.cpp:505]: nice switch case

**Review by Odycious — APPROVED** [review]

**Pull request merged**